src/logic/newTransactionPool.ts

```diff
@@ -0,0 +1,221 @@
+import {Account, Transaction, transactionSortFunc, TransactionStatus} from "src/helpers/types";
+import * as constants from 'src/helpers/constants';
+import * as AccountsSql from 'src/sql/accounts.js'
+import {getOrCreateAccount} from "src/helpers/account.utils";
+
+declare class TransactionPoolScope {
+    logger: any;
+    transactionLogic: any;
+    db: any;
+}
+
+class TransactionPool {
+
+    pool: { [transactionId: string]: Transaction } = {};
+
+    poolByRecipient: { [recipientId: string]: Array<Transaction> } = {};
+    poolBySender: { [senderId: string]: Array<Transaction> } = {};
+
+
+    scope: TransactionPoolScope = {} as TransactionPoolScope;
+
+    constructor({ transactionLogic, logger, db }: TransactionPoolScope) {
+        this.scope.transactionLogic = transactionLogic;
+        this.scope.logger = logger;
+        this.scope.db = db;
+    }
+
+    /**
+     * used in:
+     *  new transaction create on this node and verified as correct
+     *
+     * lock on moment block generation
+     */
+    set(trs: Transaction) {
+        trs.status = TransactionStatus.PUT_IN_PUL;
+        this.pool[trs.id] = trs;
+        if (!this.poolBySender[trs.senderId]) {
+            this.poolBySender[trs.senderId] = [];
+        }
+        this.poolBySender[trs.senderId].push(trs);
+        if (trs.recipientId) {
+            if (!this.poolByRecipient[trs.recipientId]) {
+                this.poolByRecipient[trs.recipientId] = [];
+            }
+            this.poolByRecipient[trs.recipientId].push(trs);
+        }
+        // TODO on this place may be broadcast logic
+    }
+
+    /**
+     * used in:
+     *  expire time
+     *  more then limit (bubble logic or not?)
+     *  put into block
+     *  exist in new common block
+     *  after verify on new trs
+     */
+    remove(trs: Transaction) {
+        delete this.pool[trs.id];
+        this.poolBySender[trs.senderId].splice(this.poolBySender[trs.senderId].indexOf(trs), 1) ;
+        this.poolByRecipient[trs.recipientId].splice(this.poolByRecipient[trs.recipientId].indexOf(trs), 1) ;
+    }
+
+    /**
+     * used in:
+     *  broadcast
+     */
+    get(id: string): Transaction {
+        return this.pool[id];
+    }
+
+    /**
+     * used in:
+     *  put into block
+     */
+    pop(trs: Transaction): Transaction {
+        const deletedValue = this.get(trs.id);
+        this.remove(trs);
+        return deletedValue;
+    }
+
+    async getUnconfirmedTransactionsForBlockGeneration(): Promise<Array<Transaction>> {
```

Comment:
Transaction pool should not know about a block generator. Please, rename the method.

Reply:
Nope, of course we need use logic like a pure components. But i think i write transaction pool for BLOCKCHAIN and it good and clear name. Because pool logically depend of block generation.